### DISCUSSION TITLE
Cannot build current v4.0.0 from source

### DISCUSSION THREAD

**Discussion by JezerM** [utmapp/UTM#4303] [General]
**Describe the issue**  

UTM v4.0.0 build process does not succeed in macOS 12.5 and Xcode 13.4.1 due to some missing classes from macOS 13 like `VZLinuxRosettaDirectoryShare` and QEMU protocols being misused. Do I require macOS 13 to build UTM v4? If so, I would recommend to document it; either, what can I do to build UTM v4.0.0 successfully?

I successfully built UTM v3.2.4 from source using sysroot-macOS-arm64 from v3.2.4 and v4.0.0, but UTM v4.0.0 didn't; also tried with recent sysroots from v4.0.1, but no result, hence it doesn't look like a sysroot error.

**Configuration**  
* UTM Version: v4.0.0
* macOS Version: 12.5
* Mac Chip (Intel, M1, ...): M1

**Build log**  

[UTM Build log.txt](https://github.com/utmapp/UTM/files/9340342/UTM.Build.log.txt)

**Comment by ktprograms** [comment]
The code referencing `VZLinuxRosettaDirectoryShare` is in a macOS 13-only code block, but Xcode 14 is required to build it (according to osy, the dev) since it still references those classes.

**Reply by JezerM** [reply]
Oh, thanks for the answer~

**Reply by apinske** [reply]
For a workaround see: https://github.com/apinske/virt/blob/5479ff2e94f440bf513e4c98f129d0190033ef7d/virt/virt/main.swift#L74

**Reply by osy** [reply]
No that won’t work. We depend on swift 5.7 language features elsewhere. You’ll get more compiler errors even if you patch out that bit.
